library/memory_buffer_alloc.c

```diff
@@ -602,6 +605,11 @@ void mbedtls_memory_buffer_alloc_init( unsigned char *buf, size_t len )
 void mbedtls_memory_buffer_alloc_free( void )
 {
 #if defined(MBEDTLS_THREADING_C)
+    if ( previous_calloc != NULL )
+    {
```

Comment:
this module is the one that uses the mbedtls_platform_set_calloc_free to modify system behavior so it should also restore the behavior otherwise mbed will fail to process any additional memory requests, see above line 576.